diff --git a/megatron/arguments.py b/megatron/arguments.py
index 5dfbb20..64ec1ae 100644
--- a/megatron/arguments.py
+++ b/megatron/arguments.py
@@ -29,6 +29,7 @@ def parse_args(extra_args_provider=None, ignore_unknown_args=False):
     parser = _add_logging_args(parser)
     parser = _add_inference_args(parser)
     parser = _add_transformer_engine_args(parser)
+    parser = _add_nlp_model_args(parser)
 
     # Custom arguments.
     if extra_args_provider is not None:
@@ -1109,3 +1110,11 @@ def _add_vision_args(parser):
                        help='warmup teacher temperaure epochs')
 
     return parser
+
+
+def _add_nlp_model_args(parser):
+    group = parser.add_argument_group(title='nlp_model')
+    group.add_argument('--nlp-model-type', type=str, default=None,
+                       choices=['bert', 'gpt', 't5'],
+                       help='NLP model type')
+    return parser
diff --git a/megatron/fused_kernels/__init__.py b/megatron/fused_kernels/__init__.py
index 7646ddb..e8e3e89 100644
--- a/megatron/fused_kernels/__init__.py
+++ b/megatron/fused_kernels/__init__.py
@@ -25,7 +25,7 @@ def load(args):
         cc_flag.append('arch=compute_80,code=sm_80')
         if int(bare_metal_minor) >= 7:
             cc_flag.append('-gencode')
-            cc_flag.append('arch=compute_90,code=sm_90')
+            cc_flag.append('arch=compute_70,code=sm_70')
 
     # Build path
     srcpath = pathlib.Path(__file__).parent.absolute()
diff --git a/setup.py b/setup.py
index c5b18c1..2a14ec2 100644
--- a/setup.py
+++ b/setup.py
@@ -1,10 +1,19 @@
 from setuptools import setup, find_packages
 
 setup(
-    name="megatron.core",
-    version="0.1",
+    name="megatron",
+    version="0.1.0",
     description="Core components of Megatron.",
-    packages=find_packages(
-        include=("megatron.core")
-    )
+    install_requires=[
+        'nltk>=3.8.1',
+        'tqdm>=4.64.1',
+        'pybind11>=2.10.3',
+        'torch>=1.13.1',
+        'Ninja>=1.11.1',
+        'flash-attn>=0.2.8',
+        'einops>=0.6.0',
+    ],
+    packages=find_packages(include=("megatron.*"), exclude=["tests"]),
+    package_data={'': ['*/Makefile', '*.h', '*.cpp', '*.cu', 'LICENSE']},
+    include_package_data=True,
 )
